local-volume/provisioner/pkg/discovery/discovery.go

```diff
@@ -255,12 +274,16 @@ func (d *Discoverer) discoverVolumesAtPath(class string, config common.MountConf
 				glog.Errorf("Path %q fs stats error: %v", filePath, err)
 				continue
 			}
+			if len(mountOptions) != 0 {
+				glog.Warningf("Path %q is already in filesystem mode, "+
```

Comment:
When we do the bind mount for filesystem mode, could you actually specify different mount options?

Reply:
Oh, thanks for pointing this out. I was mainly focus on the change this time, and did not think so thoroughly.

Then I wonder why we don't support mount options before, since we can also have the requirement in the bind mount scenario.

Reply:
Hmm, I think it's because the mount options here is kind of different from that of others.
It is not used in `MountDevice` or `Provision` phase.

Then I think it worth considering if users do have practical requirement for this, though it's supported to specify mount options in bind mount theoretically.
I see other kinds of storages did not take mount options into account in `SetUp`

Reply:
This PR updated all the volume plugins to take the same mount options for both MountDevice and SetUp: https://github.com/kubernetes/kubernetes/pull/68945

Reply:
The fact that all volume plugins do not pass mount options when calling bind was an oversight - https://github.com/kubernetes/kubernetes/pull/68945 which we fixed.

I know it is tricky for local volumes. We know that, file system mount options are ignored by bind mount and only userspace mount options do get applied. local volumes may not need too many userspace mount options so we might be safe.